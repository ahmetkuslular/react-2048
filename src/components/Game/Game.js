import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

import Cell from 'components/Cell';

Game.propTypes = {
  cells: PropTypes.array.isRequired,
  size: PropTypes.number,
  theme: PropTypes.string,
};

Game.defaultProps = {
  size: 4,
  theme: 'light',
};

function BackgroundGrid({ size }) {
  return (
    <Grid size={size}>
      {Array.from(new Array(size * size), (_, i) => i).map(i => (
        <Cell key={i} />
      ))}
    </Grid>
  );
}
function Game({ cells, size }) {
  return (
    <Wrapper>
      <BackgroundGrid size={size} />
      <Playground size={size}>
        {cells.map((row, y) =>
          row.map((value, x) => {
            if (value === 0) return null;
            return (
              <Cell key={`key_${x}_${y}`} x={x + 1} y={y + 1} value={value}>
                {value}
              </Cell>
            );
          }),
        )}
      </Playground>
    </Wrapper>
  );
}

const Wrapper = styled.div`
  border-radius: 5px;
  background-color: ${props => props.theme.primaryColor};
  height: 380px;
  width: 380px;
  padding: 10px;
`;

const Grid = styled.div`
  height: 380px;
  width: 380px;
  display: grid;
  grid-gap: 10px;
  grid-template-columns: repeat(${({ size }) => size}, 1fr);
  grid-template-rows: repeat(${({ size }) => size}, 1fr);
  position: absolute;
`;

const Playground = styled(Grid)`
  background-color: transparent;
`;

export default Game;
