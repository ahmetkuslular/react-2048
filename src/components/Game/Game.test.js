import React from 'react';
import ReactDOM from 'react-dom';
import { render } from 'enzyme';
import toJson from 'enzyme-to-json';

import Game from './Game';

const mockCells = [[0, 0, 0, 2], [0, 0, 0, 0], [0, 0, 0, 0], [0, 4, 0, 0]];
describe('components/GameSettings', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Game size={4} cells={mockCells} />, div);
  });

  //TODO: correctly testing
});
