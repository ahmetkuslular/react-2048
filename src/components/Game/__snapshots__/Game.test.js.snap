// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/GameSettings renders markup correctly for Reports 1`] = `
<div
  class="sc-bwzfXH fsgDBc"
>
  <div
    class="sc-htpNat hzRyGt"
    size="4"
  >
    <div
      class="sc-bdVaJa kreWvG"
    />
    <div
      class="sc-bdVaJa kreWvG"
    />
    <div
      class="sc-bdVaJa kreWvG"
    />
    <div
      class="sc-bdVaJa kreWvG"
    />
    <div
      class="sc-bdVaJa kreWvG"
    />
    <div
      class="sc-bdVaJa kreWvG"
    />
    <div
      class="sc-bdVaJa kreWvG"
    />
    <div
      class="sc-bdVaJa kreWvG"
    />
    <div
      class="sc-bdVaJa kreWvG"
    />
    <div
      class="sc-bdVaJa kreWvG"
    />
    <div
      class="sc-bdVaJa kreWvG"
    />
    <div
      class="sc-bdVaJa kreWvG"
    />
    <div
      class="sc-bdVaJa kreWvG"
    />
    <div
      class="sc-bdVaJa kreWvG"
    />
    <div
      class="sc-bdVaJa kreWvG"
    />
    <div
      class="sc-bdVaJa kreWvG"
    />
  </div>
  <div
    class="sc-htpNat sc-bxivhb eSgRPr"
    size="4"
  >
    <div
      class="sc-bdVaJa kreWvG"
      style="grid-column:4/5;grid-row:1/2;font-size:55px"
      value="2"
      x="4"
      y="1"
    >
      2
    </div>
    <div
      class="sc-bdVaJa kreWvG"
      style="grid-column:2/3;grid-row:4/5;font-size:55px"
      value="4"
      x="2"
      y="4"
    >
      4
    </div>
  </div>
</div>
`;
